setOptions after creation and more specific types

PabloSzx opened the pull request:
When I was using this library (very useful, thank you very much) I realized I had to make a new instance of the test client every time I had to change any option, so I made this change, and it works perfectly, as I tested it in [my fork](https://www.npmjs.com/package/@pablosz/apollo-server-integration-testing).

Also, I found a couple of easy type safety improvements to be made.

vitorbal commented:
Thanks, this makes sense to me! Could you also add some documentation to the README so folks know about this feature? Thanks in advance!

PabloSzx commented:
@vitorbal I added documentation, updated dependencies and moved TypeScript to devDependencies.

I also mentioned [graphql-tag-ts](https://www.npmjs.com/package/graphql-tag-ts) (module augmentation of [graphql-tag](https://github.com/apollographql/graphql-tag)) in the documentation since I added support for this library.

vitorbal commented:
@PabloSzx thanks for the docs, they look very good!

If it's okay by you, I'd prefer if we keep the typescript section out of this PR, to keep the docs focused on the purpose of the library.

I took a look at `graphql-tag-ts` though and it looks really cool! I'm going to try it out at work once I have a chance 🙂

PabloSzx commented:
@vitorbal done

vitorbal commented:
Published as `apollo-server-integration-testing@2.3.0`. Thanks again for your contribution, @PabloSzx!
